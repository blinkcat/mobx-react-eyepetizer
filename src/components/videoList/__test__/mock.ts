export const data = {
  date: '- Dec. 08 -',
  videoList: [
    {
      vid: 65004,
      title: '特条 |「侏罗纪世界 2：失落王国」正式预告',
      meta: '#预告 / 2\' 23"',
      cover:
        'http://img.kaiyanapp.com/7842f580b88364c1b6b637c21ced6dd0.jpeg?imageMogr2/quality/60/format/jpg'
    },
    {
      vid: 64802,
      title: '去冰岛！「来自天堂的泪滴」',
      meta: '#旅行 / 3\' 44"',
      cover:
        'http://img.kaiyanapp.com/bfdba25398e5d64b4f73fe177e9fc2f1.jpeg?imageMogr2/quality/60/format/jpg'
    },
    {
      vid: 59434,
      title: '「最诱惑」安检体验，够胆你就来',
      meta: '#剧情 / 7\' 52"',
      cover:
        'http://img.kaiyanapp.com/9a4087c122f3fed57739eb92a56db019.jpeg?imageMogr2/quality/60/format/jpg'
    },
    {
      vid: 28522,
      title: '用一部车，化解一场「世界大战」',
      meta: '#广告 / 1\' 00"',
      cover:
        'http://img.kaiyanapp.com/49fde7b936199382a651ce93d25a1438.png?imageMogr2/quality/60/format/jpg'
    },
    {
      vid: 63497,
      title: '今晚不回家，和他们一起玩极限运动吧！',
      meta: '#运动 / 9\' 00"',
      cover:
        'http://img.kaiyanapp.com/89a71472972700945d8814c4064069b3.jpeg?imageMogr2/quality/60/format/jpg'
    },
    {
      vid: 64805,
      title: '2017 冬季最「暖」广告：「朋友的臂弯」',
      meta: '#动画 / 2\' 00"',
      cover:
        'http://img.kaiyanapp.com/e84b8613eac8b64f6e97dce8bef38be3.png?imageMogr2/quality/60/format/jpg'
    }
  ]
};
