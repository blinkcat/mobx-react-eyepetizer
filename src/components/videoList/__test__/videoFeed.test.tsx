import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { VideoFeed } from '../';
import { data } from './mock';

describe('VideoFeed Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<VideoFeed {...data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
