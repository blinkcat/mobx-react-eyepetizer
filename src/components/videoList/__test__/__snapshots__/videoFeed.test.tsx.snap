// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoFeed Component renders correctly 1`] = `
<div
  className="video-list"
>
  <div
    className="date"
  >
    - Dec. 08 -
  </div>
  <div
    className="video-card"
    onClick={[Function]}
  >
    <div
      className="cover"
    />
    <div
      className="title"
    >
      特条 |「侏罗纪世界 2：失落王国」正式预告
    </div>
    <div
      className="meta"
    >
      #预告 / 2' 23"
    </div>
    <style
      jsx={true}
    >
      
        .video-card {
          background-position: 50%;
          background-size: cover;
          color: #fff;
          cursor: pointer;
          height: 450px;
          position: relative;
          background-image: url(http://img.kaiyanapp.com/7842f580b88364c1b6b637c21ced6dd0.jpeg?imageMogr2/quality/60/format/jpg);
          .title,
          .meta {
            position: absolute;
            top: 50%;
            width: 100%;
          }
          &:hover {
            .cover {
              background: transparent;
            }
          }
          @media screen and (max-width: 1024px) {
            height: 225px;
          }
          .cover {
            background: rgba(0, 0, 0, 0.4);
            height: 100%;
            position: absolute;
            transition: background 0.6s;
            width: 100%;
          }
          .title {
            font-size: 16px;
            font-weight: 700;
            transform: translateY(-24px);
          }
          .meta {
            transform: translateY(12px);
          }
        }
      
    </style>
    <style
      jsx={true}
    >
      
        .video-card-related {
          margin-bottom: 20px;
          position: relative;
          .cover {
            background-position: 50%;
            background-size: cover;
            border: 1px solid hsla(0, 0%, 100%, 0.3);
            padding-top: 25%;
            width: 40%;
            background-image: url(http://img.kaiyanapp.com/7842f580b88364c1b6b637c21ced6dd0.jpeg?imageMogr2/quality/60/format/jpg);
            @media (-webkit-min-device-pixel-ratio: 2) {
              border-width: 0.5px;
            }
          }
          .meta {
            left: 40%;
            margin-left: 15px;
            position: absolute;
            top: 50%;
            transform: translateY(-50%);
          }
          .title {
            font-size: 13px;
            font-weight: 700;
            margin-bottom: 5px;
            @media (min-width: 400px) {
              font-size: 14px;
            }
          }
          .category {
            font-size: 12px;
            opacity: 0.85;
            @media (min-width: 400px) {
              font-size: 13px;
            }
          }
        }
      
    </style>
  </div>
  <div
    className="video-card"
    onClick={[Function]}
  >
    <div
      className="cover"
    />
    <div
      className="title"
    >
      去冰岛！「来自天堂的泪滴」
    </div>
    <div
      className="meta"
    >
      #旅行 / 3' 44"
    </div>
    <style
      jsx={true}
    >
      
        .video-card {
          background-position: 50%;
          background-size: cover;
          color: #fff;
          cursor: pointer;
          height: 450px;
          position: relative;
          background-image: url(http://img.kaiyanapp.com/bfdba25398e5d64b4f73fe177e9fc2f1.jpeg?imageMogr2/quality/60/format/jpg);
          .title,
          .meta {
            position: absolute;
            top: 50%;
            width: 100%;
          }
          &:hover {
            .cover {
              background: transparent;
            }
          }
          @media screen and (max-width: 1024px) {
            height: 225px;
          }
          .cover {
            background: rgba(0, 0, 0, 0.4);
            height: 100%;
            position: absolute;
            transition: background 0.6s;
            width: 100%;
          }
          .title {
            font-size: 16px;
            font-weight: 700;
            transform: translateY(-24px);
          }
          .meta {
            transform: translateY(12px);
          }
        }
      
    </style>
    <style
      jsx={true}
    >
      
        .video-card-related {
          margin-bottom: 20px;
          position: relative;
          .cover {
            background-position: 50%;
            background-size: cover;
            border: 1px solid hsla(0, 0%, 100%, 0.3);
            padding-top: 25%;
            width: 40%;
            background-image: url(http://img.kaiyanapp.com/bfdba25398e5d64b4f73fe177e9fc2f1.jpeg?imageMogr2/quality/60/format/jpg);
            @media (-webkit-min-device-pixel-ratio: 2) {
              border-width: 0.5px;
            }
          }
          .meta {
            left: 40%;
            margin-left: 15px;
            position: absolute;
            top: 50%;
            transform: translateY(-50%);
          }
          .title {
            font-size: 13px;
            font-weight: 700;
            margin-bottom: 5px;
            @media (min-width: 400px) {
              font-size: 14px;
            }
          }
          .category {
            font-size: 12px;
            opacity: 0.85;
            @media (min-width: 400px) {
              font-size: 13px;
            }
          }
        }
      
    </style>
  </div>
  <div
    className="video-card"
    onClick={[Function]}
  >
    <div
      className="cover"
    />
    <div
      className="title"
    >
      「最诱惑」安检体验，够胆你就来
    </div>
    <div
      className="meta"
    >
      #剧情 / 7' 52"
    </div>
    <style
      jsx={true}
    >
      
        .video-card {
          background-position: 50%;
          background-size: cover;
          color: #fff;
          cursor: pointer;
          height: 450px;
          position: relative;
          background-image: url(http://img.kaiyanapp.com/9a4087c122f3fed57739eb92a56db019.jpeg?imageMogr2/quality/60/format/jpg);
          .title,
          .meta {
            position: absolute;
            top: 50%;
            width: 100%;
          }
          &:hover {
            .cover {
              background: transparent;
            }
          }
          @media screen and (max-width: 1024px) {
            height: 225px;
          }
          .cover {
            background: rgba(0, 0, 0, 0.4);
            height: 100%;
            position: absolute;
            transition: background 0.6s;
            width: 100%;
          }
          .title {
            font-size: 16px;
            font-weight: 700;
            transform: translateY(-24px);
          }
          .meta {
            transform: translateY(12px);
          }
        }
      
    </style>
    <style
      jsx={true}
    >
      
        .video-card-related {
          margin-bottom: 20px;
          position: relative;
          .cover {
            background-position: 50%;
            background-size: cover;
            border: 1px solid hsla(0, 0%, 100%, 0.3);
            padding-top: 25%;
            width: 40%;
            background-image: url(http://img.kaiyanapp.com/9a4087c122f3fed57739eb92a56db019.jpeg?imageMogr2/quality/60/format/jpg);
            @media (-webkit-min-device-pixel-ratio: 2) {
              border-width: 0.5px;
            }
          }
          .meta {
            left: 40%;
            margin-left: 15px;
            position: absolute;
            top: 50%;
            transform: translateY(-50%);
          }
          .title {
            font-size: 13px;
            font-weight: 700;
            margin-bottom: 5px;
            @media (min-width: 400px) {
              font-size: 14px;
            }
          }
          .category {
            font-size: 12px;
            opacity: 0.85;
            @media (min-width: 400px) {
              font-size: 13px;
            }
          }
        }
      
    </style>
  </div>
  <div
    className="video-card"
    onClick={[Function]}
  >
    <div
      className="cover"
    />
    <div
      className="title"
    >
      用一部车，化解一场「世界大战」
    </div>
    <div
      className="meta"
    >
      #广告 / 1' 00"
    </div>
    <style
      jsx={true}
    >
      
        .video-card {
          background-position: 50%;
          background-size: cover;
          color: #fff;
          cursor: pointer;
          height: 450px;
          position: relative;
          background-image: url(http://img.kaiyanapp.com/49fde7b936199382a651ce93d25a1438.png?imageMogr2/quality/60/format/jpg);
          .title,
          .meta {
            position: absolute;
            top: 50%;
            width: 100%;
          }
          &:hover {
            .cover {
              background: transparent;
            }
          }
          @media screen and (max-width: 1024px) {
            height: 225px;
          }
          .cover {
            background: rgba(0, 0, 0, 0.4);
            height: 100%;
            position: absolute;
            transition: background 0.6s;
            width: 100%;
          }
          .title {
            font-size: 16px;
            font-weight: 700;
            transform: translateY(-24px);
          }
          .meta {
            transform: translateY(12px);
          }
        }
      
    </style>
    <style
      jsx={true}
    >
      
        .video-card-related {
          margin-bottom: 20px;
          position: relative;
          .cover {
            background-position: 50%;
            background-size: cover;
            border: 1px solid hsla(0, 0%, 100%, 0.3);
            padding-top: 25%;
            width: 40%;
            background-image: url(http://img.kaiyanapp.com/49fde7b936199382a651ce93d25a1438.png?imageMogr2/quality/60/format/jpg);
            @media (-webkit-min-device-pixel-ratio: 2) {
              border-width: 0.5px;
            }
          }
          .meta {
            left: 40%;
            margin-left: 15px;
            position: absolute;
            top: 50%;
            transform: translateY(-50%);
          }
          .title {
            font-size: 13px;
            font-weight: 700;
            margin-bottom: 5px;
            @media (min-width: 400px) {
              font-size: 14px;
            }
          }
          .category {
            font-size: 12px;
            opacity: 0.85;
            @media (min-width: 400px) {
              font-size: 13px;
            }
          }
        }
      
    </style>
  </div>
  <div
    className="video-card"
    onClick={[Function]}
  >
    <div
      className="cover"
    />
    <div
      className="title"
    >
      今晚不回家，和他们一起玩极限运动吧！
    </div>
    <div
      className="meta"
    >
      #运动 / 9' 00"
    </div>
    <style
      jsx={true}
    >
      
        .video-card {
          background-position: 50%;
          background-size: cover;
          color: #fff;
          cursor: pointer;
          height: 450px;
          position: relative;
          background-image: url(http://img.kaiyanapp.com/89a71472972700945d8814c4064069b3.jpeg?imageMogr2/quality/60/format/jpg);
          .title,
          .meta {
            position: absolute;
            top: 50%;
            width: 100%;
          }
          &:hover {
            .cover {
              background: transparent;
            }
          }
          @media screen and (max-width: 1024px) {
            height: 225px;
          }
          .cover {
            background: rgba(0, 0, 0, 0.4);
            height: 100%;
            position: absolute;
            transition: background 0.6s;
            width: 100%;
          }
          .title {
            font-size: 16px;
            font-weight: 700;
            transform: translateY(-24px);
          }
          .meta {
            transform: translateY(12px);
          }
        }
      
    </style>
    <style
      jsx={true}
    >
      
        .video-card-related {
          margin-bottom: 20px;
          position: relative;
          .cover {
            background-position: 50%;
            background-size: cover;
            border: 1px solid hsla(0, 0%, 100%, 0.3);
            padding-top: 25%;
            width: 40%;
            background-image: url(http://img.kaiyanapp.com/89a71472972700945d8814c4064069b3.jpeg?imageMogr2/quality/60/format/jpg);
            @media (-webkit-min-device-pixel-ratio: 2) {
              border-width: 0.5px;
            }
          }
          .meta {
            left: 40%;
            margin-left: 15px;
            position: absolute;
            top: 50%;
            transform: translateY(-50%);
          }
          .title {
            font-size: 13px;
            font-weight: 700;
            margin-bottom: 5px;
            @media (min-width: 400px) {
              font-size: 14px;
            }
          }
          .category {
            font-size: 12px;
            opacity: 0.85;
            @media (min-width: 400px) {
              font-size: 13px;
            }
          }
        }
      
    </style>
  </div>
  <div
    className="video-card"
    onClick={[Function]}
  >
    <div
      className="cover"
    />
    <div
      className="title"
    >
      2017 冬季最「暖」广告：「朋友的臂弯」
    </div>
    <div
      className="meta"
    >
      #动画 / 2' 00"
    </div>
    <style
      jsx={true}
    >
      
        .video-card {
          background-position: 50%;
          background-size: cover;
          color: #fff;
          cursor: pointer;
          height: 450px;
          position: relative;
          background-image: url(http://img.kaiyanapp.com/e84b8613eac8b64f6e97dce8bef38be3.png?imageMogr2/quality/60/format/jpg);
          .title,
          .meta {
            position: absolute;
            top: 50%;
            width: 100%;
          }
          &:hover {
            .cover {
              background: transparent;
            }
          }
          @media screen and (max-width: 1024px) {
            height: 225px;
          }
          .cover {
            background: rgba(0, 0, 0, 0.4);
            height: 100%;
            position: absolute;
            transition: background 0.6s;
            width: 100%;
          }
          .title {
            font-size: 16px;
            font-weight: 700;
            transform: translateY(-24px);
          }
          .meta {
            transform: translateY(12px);
          }
        }
      
    </style>
    <style
      jsx={true}
    >
      
        .video-card-related {
          margin-bottom: 20px;
          position: relative;
          .cover {
            background-position: 50%;
            background-size: cover;
            border: 1px solid hsla(0, 0%, 100%, 0.3);
            padding-top: 25%;
            width: 40%;
            background-image: url(http://img.kaiyanapp.com/e84b8613eac8b64f6e97dce8bef38be3.png?imageMogr2/quality/60/format/jpg);
            @media (-webkit-min-device-pixel-ratio: 2) {
              border-width: 0.5px;
            }
          }
          .meta {
            left: 40%;
            margin-left: 15px;
            position: absolute;
            top: 50%;
            transform: translateY(-50%);
          }
          .title {
            font-size: 13px;
            font-weight: 700;
            margin-bottom: 5px;
            @media (min-width: 400px) {
              font-size: 14px;
            }
          }
          .category {
            font-size: 12px;
            opacity: 0.85;
            @media (min-width: 400px) {
              font-size: 13px;
            }
          }
        }
      
    </style>
  </div>
  <style
    jsx={true}
  >
    
        .video-list {
          text-align: center;
          .date {
            background: #fff;
            font-family: Lobster, cursive;
            font-size: 16px;
            line-height: 90px;
          }
        }
      
  </style>
</div>
`;
